DOCUMENTATION.md

```diff
@@ -0,0 +1,137 @@
+Bluetooth | `--allow=bluetooth` and `--disallow=bluetooth` | Toggle | Permit (`--allow=bluetooth`) or prohibit (`--disallow=bluetooth`) access to the application to use bluetooth.
+Controller Area Network bus | `--allow=canbus` and `--disallow=canbus` | Toggle | Permit (`--allow=canbus`) or prohibit (`--disallow=canbus`) access to the application to use canbus sockets. You must also have network access for this to work.
+
+#### Filesystem
+
+List of filesystem subsets available to the application.
+
+Name | `flatpak override` equivalent | Type | Description
+--- | --- | --- | ---
+All filesystem files | `--filesystem=host` and `--nofilesystem=host` | Toggle | Permit (`--filesystem=host`) or prohibit (`--nofilesystem=host`) read-write access to the application to the whole filesystem. If it is permitted, the application has read-write access to every file and folder owned by you, the user. The rest will be read-only.
+All system libraries, executables and static data | `--filesystem=host-os` and `--nofilesystem=host-os` | Toggle | Permit (`--filesystem=host-os`) or prohibit (`--nofilesystem=host-os`) read-write access to the application to system libraries located in `/usr`. Since this directory requires root access to write, the permission will be read-only.
+All system configurations | `--filesystem=host-etc` and `--nofilesystem=host-etc` | Toggle | Permit (`--filesystem=host-etc`) or prohibit (`--nofilesystem=host-etc`) read-write access to the application to system configurations located in `/etc`. Since this directory requires root access to write, the permission will be read-only.
+All user files | `--filesystem=home` and `--nofilesystem=home` | Toggle | Permit (`filesystem=home`) or prohibit (`--nofilesystem=home`) read-write access to the application to the user directory (`$HOME`).
+Other files | `--filesystem=[PATH]`, `--filesystem=[PATH]:ro` and `--nofilesystem=[PATH]` | Input | Permit (`--filesystem=[PATH]`) or prohibit `--nofilesystem=[PATH]` read-write access to the application to the directory you desire, for example in Flatseal, you would put `~/games` (`--filesystem=~/games` using `flatpak override`) in the box if you want read-write access to `~/games`. If you want read-only access (`ro`), then you want append `:ro`, for example in Flatseal, you would put `~/games:ro` (`--filesystem=~/games:ro` using `flatpak override`) if you want read-only access to `~/games`.
+
+#### Persistent
+
+List of the homedir-relative paths created in the sandbox.
+
+Name | `flatpak-override` equivalent | Type | Description
+--- | --- | --- | ---
+Files | `--persist=[FILENAME]` | Input | If the application doesn't have access to the real homedir, make the (homedir-relative) path FILENAME a bind mount to the corresponding path in the per-application directory, allowing that location to be used for persistent data. This updates the [Context] group in the metadata. This option can be used multiple times. 
```

Comment:
This is one of the rare cases where the Flatpak doc is really hard to understand. What persistent does is

1. Creates a directory, relative to the home directory, but
2. Makes this directory visible only to the application, when the application is running (it doesn't really exists outside of the sandbox).
3. Makes sure that whatever is written in that directory will persistent, this means that the contents will be available next time the apps run.
3.

Reply:
The actual directory is created and does exist in ~/.var/app/ID but, when the app is running, inside the sandbox, it shows as if it was placed directly under `$HOME`.

Reply:
That's really intelligent and well made, but how would we write that in a way that is very easy to understand?

Reply:
haha, I suppose that is the reason why the Flatpak version of that description ins't easy to understand either...

I could be something like "Creates a persistent path under the $HOME directory, only available to the application" ?

Reply:
It's not really a "path under the $HOME directory" because it remaps the wanted directory to `~/.var/app/$ID/`.

Reply:
yes, but that's the kind of implementation detail that is irrelevant to someone wanting to use Flatseal. They just need to understand _what_ it does, not _how_. Because these details are so convoluted that makes the docs useless for regular users.

Reply:
I agree. But should we say the "what" first and then the "how"?

Reply:
I think this is one of those cases where it's better to give a practical example instead of a "complete" explanation. Example:

e.g. instead of giving zoom access a the real "~/.zoom" directory, what `--pesist` does is to provide a fake "~/.zoom" directory that is only available to the application. 

something like that,

Reply:
The problem is that if one uses persist, some might think that `~/.zoom` will still be in use, when it reality it will be in `~/.var/app/us.zoom.Zoom/.zoom`. And even worse, persist only works when the program has no write access to the targetted directory.

Reply:
Yeah, but we can always point to the Flatpak more complete docs. The goal should be to make Flatseal easier to use and understand first, and then we can point to more complete technical description.

Otherwise is no different from the Flatpak docs.

Reply:
I would say something like:

> The targeted directory will be remapped to the Flatpak program's directory (`~/.var/app/$FLATPAK_APP_ID/[DIRECTORY]`) **if it has no write access to the targeted directory**.
> 
> For example, persisting `.mozilla` will map `~/.mozilla` to `~/.var/app/org.mozilla.Firefox/.mozilla` if the Flatpak has no write access to `~/.mozilla`.

Reply:
> Yeah, but we can always point to the Flatpak more complete docs. The goal should be to make Flatseal easier to use and understand first, and then we can point to more complete technical description.

Fair. But it's good to make it clear because the Flatpak docs aren't clear, in my opinion.

Reply:
Also, what are your thoughts on https://github.com/tchx84/Flatseal/pull/136#discussion_r498863589?

Reply:
Yeah, is fine... but it's focusing on the how, not the what. It should, at least, explain why one would do such a thing (e.g. the zoom example).

Reply:
> This is used to make a directory only accessible by a Flatpak program to prevent other Flatpak programs from reading it.
> 
> The targeted directory will be remapped to the Flatpak program's directory (`~/.var/app/$FLATPAK_APP_ID/[PATH]`) **if it has no write access to the targeted directory**.
> 
> For example, persisting `.mozilla` will map `~/.mozilla` to `~/.var/app/org.mozilla.Firefox/.mozilla` if the Firefox Flatpak has no write access to `~/.mozilla`.

Reply:
Yeah is getting there ;) and sure any example is fine.

Reply:
It should look good now https://github.com/tchx84/Flatseal/pull/136/commits/502f197dbfbbfbea643b59d0b36a57f6c96a5f00

Reply:
https://github.com/tchx84/Flatseal/pull/136/commits/fb08e338524f9957ef54a97434b4290f84020447